### PR TITLE
Memory leak in determining equality of two type id+obj pairs

### PR DESCRIPTION
**Pull request by eboasson**

### CODE REVIEW

File: src/core/ddsi/src/ddsi_typelib.c [hunk 1]
```diff
@@ -244,10 +244,26 @@ static bool ti_to_pairs_equal (const dds_sequence_DDS_XTypes_TypeIdentifierTypeO
   if (a->_length != b->_length)
     return false;
 
-  struct dds_cdrstream_desc desc;
-  dds_cdrstream_desc_from_topic_desc (&desc, &DDS_XTypes_TypeObject_desc);
-
-  for (uint32_t n = 0; n < a->_length; n++)
+  // Can't do this (yet) is as a compile time constant because we need
+  // DDS_XTypes_TypeObject_desc.m_ops and the value is not visible here
```
Comment (dpotman) [inline src/core/ddsi/src/ddsi_typelib.c:248]: I think this should be `DDS_XTypes_TypeObject_ops`

File: src/core/ddsi/src/ddsi_typelib.c [hunk 2]
```diff
@@ -244,10 +244,26 @@ static bool ti_to_pairs_equal (const dds_sequence_DDS_XTypes_TypeIdentifierTypeO
...
+      .nkeys = 0,
+      .keys = NULL
+    },
+    .ops = {
+      .nops = 0, // serializer VM doesn't used it during execution
+      .ops = (uint32_t *) DDS_XTypes_TypeObject_desc.m_ops
+    },
+    .opt_size_xcdr1 = 0,
+    .opt_size_xcdr2 = 0
+  };
```
Comment (dpotman) [inline src/core/ddsi/src/ddsi_typelib.c:263]: We could add an option in IDLC to generate a `dds_cdrstream_desc` in addition to the topic descriptor type and include this in the generated meta-data for type info/object/map in DDSI. Then we can avoid this transformation, and also for a few other instances where a type object/information/map is serialized.

**Comment by thijsmie** [comment]
You've leaked your paths to the typeinfo IDL 😇

**Comment by eboasson** [comment]
> You've leaked your paths to the typeinfo IDL 😇

I should fix that, good policy not do that — thanks ☺️

File: src/tools/idlpp/src/mcpp_out.h [hunk 3]
```diff
@@ -13,10 +13,10 @@
 #define MCPP_OUT_H
 
 typedef enum {
-  OUT,
-  ERR,
-  DBG,
+  MCPP_OUT,
+  MCPP_ERR,
+  MCPP_DBG,
   NUM_OUTDEST
```
Comment (thijsmie) [inline src/tools/idlpp/src/mcpp_out.h:19]: Not `MCPP_NUM_OUTDEST` for consistency?

Comment (eboasson) [reply src/tools/idlpp/src/mcpp_out.h:19]: Bummer, that's what you get for automating source-to-source transformations on a C code base ... This is too terrible, I'll fix this before merging it.

**Review by thijsmie — APPROVED** [review]

**Comment by eboasson** [comment]
Thanks @thijsmie ! I'm force-pushing it for squashing the two `MCPP_` prefixing comments with no further changes, so I'll merge it immediately afterward

**Pull request merged**